common/src/main/java/com/microsoft/identity/common/adal/internal/BrokerRtAccessor.java

```diff
@@ -0,0 +1,83 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.adal.internal;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.authscheme.BearerAuthenticationSchemeInternal;
+import com.microsoft.identity.common.internal.broker.BrokerValidator;
+import com.microsoft.identity.common.internal.cache.ICacheRecord;
+import com.microsoft.identity.common.internal.cache.MsalOAuth2TokenCache;
+import com.microsoft.identity.common.internal.dto.AccountRecord;
+
+import static com.microsoft.identity.common.exception.ClientException.TOKEN_CACHE_ITEM_NOT_FOUND;
+
+public final class BrokerRtAccessor {
+
+    private final MsalOAuth2TokenCache mTokenCache;
+    private final Context mContext;
+
+    public BrokerRtAccessor(@NonNull final MsalOAuth2TokenCache cache,
+                            @NonNull final Context context) {
+        mTokenCache = cache;
+        mContext = context;
+    }
+
+    public String getBrokerRt(String accountObjectId) throws ClientException {
+        if (!AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equals(mContext.getPackageName()) &&
+                !AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME.equals(mContext.getPackageName())) {
+            throw new ClientException("This can only be invoked by Broker apps.");
+        }
+
+        if (!new BrokerValidator(mContext).verifySignature(mContext.getPackageName())) {
+            throw new ClientException("This can only be invoked by apps with a valid signature hash.");
+        }
+
+        final ICacheRecord cacheRecord = getCacheRecordForIdentifier(accountObjectId);
+        return cacheRecord.getRefreshToken().getSecret();
+    }
+
+    private ICacheRecord getCacheRecordForIdentifier(@NonNull final String accountObjectId) throws ClientException {
+        final AccountRecord localAccountRecord = mTokenCache.getAccountByLocalAccountId(
+                null,
+                AuthenticationConstants.Broker.BROKER_CLIENT_ID,
+                accountObjectId
+        );
+
+        // Check it's not null
+        if (null == localAccountRecord) {
+            // Unrecognized identifier, cannot supply a token.
+            throw new ClientException(TOKEN_CACHE_ITEM_NOT_FOUND);
+        }
+
+        return mTokenCache.load(
```

Comment:
Given today's meeting, we might to to rethink this logic, as this would save the BRT in the MSAL unified cache

Reply:
The first one is clean enough to do, the second one I think it'll break the pattern we have.

Eventually, I think if we want to restrict BRT lifetime, we should be enforcing this on the server side.